html/modules/custom/hr_paragraphs/src/Controller/ParagraphController.php

```diff
@@ -320,64 +327,67 @@ public function getPages($group) {
    * Return all datasets of an operation, sector or cluster.
    */
   public function getDatasets($group, Request $request) {
-    $limit = 10;
-    $offset = 0;
+    if ($group->hasField('field_hdx_alternate_source') && !$group->field_hdx_alternate_source->isEmpty()) {
+      /** @var \Drupal\link\Plugin\Field\FieldType\LinkItem $link */
+      $link = $group->field_hdx_alternate_source->first();
 
-    if ($request->query->has('page')) {
-      $offset = $request->query->getInt('page', 0) * $limit;
+      // Redirect external links.
+      if ($link->isExternal()) {
+        return new TrustedRedirectResponse($link->getUrl()->getUri());
+      }
     }
 
-    if ($group->field_hdx_country->isEmpty()) {
+    if ($group->field_hdx_dataset_link->isEmpty()) {
```

Comment:
When I pulled develop and integrated all this, I am seeing a WSOD. When I looked at logs I got this error:

```
PHP message: Error: Call to a member function isEmpty() on null in /var/www/html/modules/custom/hr_paragraphs/src/Controller/ParagraphController.php on line 340
```

I see the block above and tried adding to the conditional:

```php
    if (!$group->hasField('field_hdx_dataset_link') || $group->field_hdx_dataset_link->isEmpty()) {
      return [
        '#type' => 'markup',
        '#markup' => $this->t('HDX country not chosen.'),
      ];
    }
```

But now I'm seeing the "HDX country not chosen" text, even though in the backend I have Afghanistan chosen. I tried removing it and re-saving the Operation.